src/CommunityToolkit.Maui/Extensions/Navigation/NavigationExtensions.shared.cs

```diff
@@ -0,0 +1,70 @@
+﻿
+using System.Threading.Tasks;
+using CommunityToolkit.Maui.Core;
+using CommunityToolkit.Maui.Views;
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Platform;
+
+namespace CommunityToolkit.Maui.Extensions;
+
+/// <summary>
+/// Extension methods for <see cref="INavigation"/>.
+/// </summary>
+public static partial class NavigationExtensions
+{
+	/// <summary>
+	/// Displays a popup.
+	/// </summary>
+	/// <param name="navigation">
+	/// The current <see cref="INavigation"/>.
+	/// </param>
+	/// <param name="popup">
+	/// The <see cref="BasePopup"/> to display.
+	/// </param>
+	public static void ShowPopup(this INavigation navigation, BasePopup popup)
+	{
+#if WINDOWS
+		PlatformShowPopup(popup, GetMauiContext(navigation));
+#else
+		CreatePopup(navigation, popup);
+#endif
+
+	}
+
+	/// <summary>
+	/// Displays a popup and returns a result.
+	/// </summary>
+	/// <param name="navigation">
+	/// The current <see cref="INavigation"/>.
+	/// </param>
+	/// <param name="popup">
+	/// The <see cref="Popup"/> to display.
+	/// </param>
+	/// <returns>
+	/// A task that will complete once the <see cref="Popup"/> is dismissed.
+	/// </returns>
+	public static Task<object?> ShowPopupAsync(this INavigation navigation, Popup popup)
+	{
+#if WINDOWS
+		return PlatformShowPopupAsync(popup, GetMauiContext(navigation));
+#else
+
+		CreatePopup(navigation, popup);
+		return popup.Result;
+#endif
+	}
+
+	static void CreatePopup(INavigation navigation, BasePopup popup)
+	{
+		var mauiContext = GetMauiContext(navigation);
+		var popupNative = popup.ToHandler(mauiContext);
+		popupNative.Invoke(nameof(IPopup.OnOpened));
+	}
+
+	static IMauiContext GetMauiContext(INavigation navigation)
+	{
+		return (Shell.Current is null ? 
+			navigation.NavigationStack[0].Handler?.MauiContext
```

Comment:
possible index out of range exception:
```
var page = new MainPage()
page.ShowPopup();
```

Reply:
This will not happen, because the user needs to have a `NavigationPage`, as you can see the `NavigationExtensions` receives an `INavigation` as a parameter. If we think that using a Page instead is better (and will work), we can change this in a future PR